test/powershell/Modules/PSDiagnostics/PSDiagnostics.Tests.ps1

```diff
@@ -0,0 +1,76 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+Describe "PSDiagnostics cmdlets tests" -Tag "CI", "RequireAdminOnWindows" {
+    BeforeAll {
+        $OriginalDefaultParameterValues = $PSDefaultParameterValues.Clone()
+        if ( -not $IsWindows ) {
+            $PSDefaultParameterValues["it:skip"] = $true
+        }
+    }
+    AfterAll {
+        $Global:PSDefaultParameterValues = $OriginalDefaultParameterValues
+    }
+
+    Context "Test for Enable-PSTrace cmdlet" {
+        it "Should enable Analytic logs for Microsoft-Windows-PowerShell." {
+            Enable-PSTrace -Force
+
+            [XML]$WevtUtilOutput = & wevtutil gl Microsoft-Windows-PowerShell/Analytic /f:xml
+
+            $WevtUtilOutput.Channel.enabled | Should -Be 'true'
+        }
+    }
+
+    Context "Test for Disable-PSTrace cmdlet" {
+        it "Should disable Analytic logs for Microsoft-Windows-PowerShell." {
+            Disable-PSTrace
+
+            [XML]$WevtUtilOutput = & wevtutil gl Microsoft-Windows-PowerShell/Analytic /f:xml
+
+            $WevtUtilOutput.Channel.enabled | Should -Be 'false'
+        }
+    }
+
+    Context "Test for Get-LogProperties cmdlet" {
+        it "Should show properties of Admin logs for 'Microsoft-Windows-PowerShell'." {
+            [XML]$WevtUtilOutput = wevtutil gl Microsoft-Windows-PowerShell/Admin /f:xml
+
+            $LogProperty = Get-LogProperties -Name Microsoft-Windows-PowerShell/Admin
+
+            $LogProperty.Name       | Should -Be $WevtUtilOutput.channel.Name
+            $LogProperty.Enabled    | Should -Be $WevtUtilOutput.channel.Enabled
+            $LogProperty.Retention  | Should -Be $WevtUtilOutput.channel.Logging.Retention
+            $LogProperty.AutoBackup | Should -Be $WevtUtilOutput.channel.Logging.AutoBackup
+            $LogProperty.MaxLogSize | Should -Be $WevtUtilOutput.channel.Logging.MaxSize
```

Comment:
Please add a tests for `$LogProperty.Count` to ensure that we get and check ALL properties.

Reply:
Didn't get you, are you referring to `($LogProperty | Get-Member -MemberType Property | Measure-Object).Count`

Reply:
Yes. If new property will be added we should update the test and add a check for new property.